sale_variant_configurator/models/sale_order.py

```diff
@@ -102,28 +103,26 @@ def _onchange_product_tmpl_id_configurator(self):
             self.name = (self.name or "") + "\n" + product_tmpl.description_sale
         if self.order_id.pricelist_id and self.order_id.partner_id:
             self.price_unit = self.env["account.tax"]._fix_tax_included_price(
-                product_tmpl.price,
+                product_tmpl.list_price,
                 product_tmpl.taxes_id,
                 self.tax_id,
             )
         return res
 
     @api.onchange("product_id")
-    def product_id_change(self):
+    def _onchange_product_id(self):
```

Comment:
You should not rename the method and create a new one, but hook into the existing one that has already been renamed.
In v15: https://github.com/odoo/odoo/blob/15.0/addons/sale/models/sale_order_line.py#L617
In v16: https://github.com/odoo/odoo/blob/16.0/addons/sale/models/sale_order_line.py#L939

Commit: https://github.com/odoo/odoo/commit/1d3b9f6be6f664eaa344028a96926f15fc59428f